packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -600,10 +600,16 @@ function build(
     return result;
   }
 
-  function registerValue(slot, val) {
+  function registerValue(slot, val, faceted) {
```

Comment:
I'm disinclined to do this, both for the reason you mention and because it would result in needless code duplication just to placate TypeScript.  TypeScript is supposed to be our servant, not the other way around.  We should discuss.

Reply:
In discussion, @warner persuaded me (or, at least, caused me to persuade myself) that separate function is the way to go.  Tentatively calling these `registerValue` and `registerValueCohort`.

Reply:
As I went to implement these changes, reality intruded and pushed to revert the previous decision -- it turns out that in the places that the `faceted` flag is used, it really is a boolean that could take either value, so each call site would have to be translated into an `if`.  Furthermore, we'd now have to pass around two functions everywhere that we currently pass around the one `registerValue` function.  Making the parameter mandatory everywhere rather than defaulting it when it is `false` also helps clarify things.